File: inspiration-space-server/inspiration-space-client/src/main/java/com/is/inspirationspaceclient/user/service/UserService.java
package com.is.inspirationspaceclient.user.service;

import com.is.inspirationspaceclient.user.model.dto.*;
import com.is.inspirationspaceclient.user.model.vo.*;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

/**
 * <p>
 * 用户表 服务类
 * </p>
 *
 * @author
 * @since 2025-07-07
 */
public interface UserService {


    Boolean register(UserRegisterDto userRegisterDto);

    UserLoginVo login(UserLoginDto userloginDto);

    UserProfileVo getUserProfile(Long userId);

    UserExtendVo getUserExtendInfo(Long userId);

    UserStatsVo getUserStatsInfo(Long userId);

    UserVo getUser(Long userId);

    Boolean updateUser(UserUpdateDto userUpdateDto);

    Boolean logout(UserIdDto userIdDto);

    Boolean updatePassword(UserChangePasswordDto userChangePasswordDto);

    Boolean removeById(UserIdDto userIdeDto);

    UserInfoVo getUserInfo(Long userId);

    String getAvatar(Long userId);

    Boolean updateAvatar(Long userId, MultipartFile avatar);

    /**
     * 关注用户
     */
    Boolean followUser(Long targetUserId);

    /**
     * 取消关注用户
     */
    Boolean unfollowUser(Long targetUserId);

    /**
     * 判断是否已关注
     */
    Boolean isFollowing(Long targetUserId);

    /**
     * 获取用户排行榜
     * @param topN 前N名
     * @return 用户列表
     */
    List<UserVo> getUserRanking(int topN);
}

File: frontend/inspiration-space/src/services/userService.js
import { publicApiCall, privateApiCall } from './apiClient';


// 公开接口
export const register = (userRegisterDto) => 
  publicApiCall('/user/register', 'POST', userRegisterDto);

export const login = (userLoginDto) => 
  publicApiCall('/user/login', 'POST', userLoginDto);

// 需要client-user角色的接口
export const getUserProfileInfo = (userId) => 
  privateApiCall(`/user/getProfile/${userId}`, 'GET');

export const getUserExtendInfo = (userId) => 
  privateApiCall(`/user/getExtend/${userId}`, 'GET');

export const getUserStatsInfo = (userId) => 
  privateApiCall(`/user/getStats/${userId}`, 'GET');

// 公开用户信息接口
export const getUser = (userId) => 
  publicApiCall(`/user/get/${userId}`, 'GET');

export const getUserInfo = (userId) => 
  publicApiCall(`/user/getUserInfo/${userId}`, 'GET');

export const getAvatar = (userId) =>
  publicApiCall(`/user/getAvatar/${userId}`, 'GET');

// 用户操作接口
export const updateUserInfo = (userUpdateDto) => 
  privateApiCall('/user/update', 'POST', userUpdateDto);

export const logout = (userIdDto) => 
  privateApiCall('/user/logout', 'POST', userIdDto);

export const updatePassword = (userChangePasswordDto) => 
  privateApiCall('/user/changePassword', 'PATCH', userChangePasswordDto);

export const deleteAccount = (userIdDto) => 
  privateApiCall('/user/delete', 'DELETE', userIdDto);

export const updateAvatar = (userId, formData) => {
  // 路径参数拼接 userId，而非 formData；请求体传递 FormData
  return privateApiCall(`/user/updateAvatar/${userId}`, 'POST', formData);
};


